lib/std/mem.zig

```diff
@@ -645,7 +645,18 @@ const backend_can_use_eql_bytes = switch (builtin.zig_backend) {
 
 /// Compares two slices and returns whether they are equal.
 pub fn eql(comptime T: type, a: []const T, b: []const T) bool {
-    if (@sizeOf(T) == 0) return true;
+    switch (@typeInfo(T)) {
+        .Type, .ComptimeInt, .ComptimeFloat => {
+            if (a.len != b.len) return false;
+            inline for (a, b) |a_elem, b_elem| {
+                if (a_elem != b_elem) return false;
+            }
+            return true;
+        },
+        .Null, .Undefined => return a.len == b.len,
+        else => {},
```

Comment:
imo this should not use `else` but otherwise lgtm

Reply:
What do you propose instead? The switch must handle all possibilities.

Reply:
still having `=> {},` but having all the individual names, there's not a ton and its more maintainable to update if the list ever changes